src/app/components/did/update-did/service-form/service-form.component.html

```diff
@@ -0,0 +1,74 @@
+<form [formGroup]="serviceForm" (ngSubmit)="addService()">
+  <div class="form-group">
+    <label class="control-label" for="type">
+      Service type
+      <i class="fas fa-info-circle fa-lg text-primary" ngbTooltip="{{typeTooltipMessage}}" tooltipClass="custom-tooltip"></i>
+    </label>
+    <input
+      type="text"
+      class="form-control input-md"
+      formControlName="type"
+      id="type"
+      placeholder="Service type"
+      [ngClass]="{ 'is-invalid': type.touched && type.errors }" />
+    <div *ngIf="type.errors" class="invalid-feedback">
+      <div *ngIf="type['errors'].required">Service type is required!</div>
+    </div>
+  </div>
+
+  <div class="form-group">
+    <label class="control-label" for="endpoint">
+      Service endpoint
+      <i class="fas fa-info-circle fa-lg text-primary" ngbTooltip="{{endpointTooltipMessage}}" tooltipClass="custom-tooltip"></i>
+    </label>
+    <input
+      type="text"
+      class="form-control input-md"
+      formControlName="endpoint"
+      id="endpoint"
+      placeholder="Service endpoint"
+      pattern="^(http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$"
```

Comment:
A general comment: I'm wondering if it makes sense to use an existing library for validation of URLs (and maybe other data as well), such as this one https://validatejs.org/#validators-url, for example. Just something to consider, as usually these libraries have very well tested and robust validators and validating URLs in particular is tricky.

Reply:
I will consider it for the next PR.